src/jinja2/lexer.py

```diff
@@ -146,17 +146,22 @@
     f"({'|'.join(re.escape(x) for x in sorted(operators, key=lambda x: -len(x)))})"
 )
 
-ignored_tokens = frozenset(
+comment_tokens = frozenset(
     [
         TOKEN_COMMENT_BEGIN,
         TOKEN_COMMENT,
         TOKEN_COMMENT_END,
-        TOKEN_WHITESPACE,
         TOKEN_LINECOMMENT_BEGIN,
         TOKEN_LINECOMMENT_END,
         TOKEN_LINECOMMENT,
     ]
 )
+ignored_tokens = frozenset(
```

Comment:
note: `ignored_tokens` kept unchanged for backward-compatibility. `comment_tokens` is later subtracted from it depending on whether comments must be preserved or not.